analytics/tracker.go

```diff
@@ -230,19 +236,33 @@ func (t tracker) SendStepFinishedEvent(properties analytics.Properties, result S
 		return
 	}
 
-	var eventName string
-	var extraProperties analytics.Properties
+	var (
+		eventName       string
+		extraProperties analytics.Properties
+	)
 
 	switch result.Status {
 	case models.StepRunStatusCodeSuccess:
 		eventName = stepFinishedEventName
 		extraProperties = analytics.Properties{statusProperty: successfulValue}
-		break
 	case models.StepRunStatusCodeFailed, models.StepRunStatusCodeFailedSkippable:
 		eventName = stepFinishedEventName
 		extraProperties = analytics.Properties{statusProperty: failedValue}
 		extraProperties.AppendIfNotEmpty(errorMessageProperty, result.ErrorMessage)
-		break
+	case models.StepRunStatusAbortedTimeout:
+		eventName = stepAbortedEventName
+		extraProperties = analytics.Properties{reasonProperty: customTimeoutValue}
+
+		if result.Timeout >= 0 {
+			extraProperties[timeoutProperty] = result.Timeout.Seconds()
+		}
+	case models.StepRunStatusAbortedNoOutputTimeout:
+		eventName = stepAbortedEventName
+		extraProperties = analytics.Properties{reasonProperty: noOutputTimeoutValue}
```

Comment:
Can we add the custom timeout value here, just like in the case of step timeout?